src/Compilers/CSharp/Portable/BoundTree/IBoundAnonymousObjectCreation.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal interface IBoundAnonymousObjectCreation
```

Comment:
What's the purpose of those IBoundX interfaces? I don't see them used anywhere

Reply:
These are used for Lazy operation nodes that need to hold onto a BoundNode that could be one of a few different types, but those types don't make sense to have a base-class relationship together. For this one, it's implemented by BoundAnonymousObjectCreationExpression and BoundObjectCreationExpression, and used by CSharpLazyAnonymousObjectCreationOperation

---
In reply to: [236830119](https://github.com/dotnet/roslyn/pull/31010#discussion_r236830119) [](ancestors = 236830119)